api/lib/domain/usecases/get-external-authentication-redirection-url.js

```diff
@@ -0,0 +1,26 @@
+module.exports = async function getExternalAuthenticationRedirectionUrl({
+  userAttributes,
+  userRepository,
+  tokenService,
+  settings,
+}) {
+  const { attributeMapping } = settings.saml;
+
+  const externalUser = {
+    firstName: userAttributes[attributeMapping.firstName],
+    lastName: userAttributes[attributeMapping.lastName],
+    samlId: userAttributes[attributeMapping.samlId],
+  };
+
+  const foundUser = await userRepository.getBySamlId(externalUser.samlId);
+
+  if (foundUser) {
+    const token = tokenService.createTokenFromUser(foundUser, 'external');
+
+    return `/?token=${encodeURIComponent(token)}&user-id=${foundUser.id}`;
+  }
+
+  const externalUserToken = tokenService.createTokenForStudentReconciliation(externalUser);
+
+  return `/campagnes?externalUser=${encodeURIComponent(externalUserToken)}`;
+};
```

Comment:
Difficile à la lecture de comprendre qu'on bascule entre deux cas bien distincts :
- Un utilisateur qui dispose déjà d'un compte, et qu'on **authentifie**
- Un utilisateur qui se connecte depuis le GAR et qui ne dispose pas encore de compte, qu'on a besoin **d'identifier**

```suggestion
  if (foundUser) {
    const token = tokenService.createTokenFromUser(foundUser, 'external');

    return `/?token=${encodeURIComponent(token)}&user-id=${foundUser.id}`;
  } else {
  
    const externalUserToken = tokenService.createTokenForStudentReconciliation(externalUser);

    return `/campagnes?externalUser=${encodeURIComponent(externalUserToken)}`;
  }
};
```

Reply:
Done